src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Input, OnInit, forwardRef, ViewChild } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { SelectItem } from 'primeng/primeng';
+import { AceEditorComponent } from 'ng2-ace-editor';
+
+@Component({
+    selector: 'dot-textarea-content',
+    templateUrl: './dot-textarea-content.component.html',
+    styleUrls: ['./dot-textarea-content.component.scss'],
+    providers: [
+        {
+            multi: true,
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => DotTextareaContentComponent)
+        }
+    ]
+})
+export class DotTextareaContentComponent implements OnInit, ControlValueAccessor {
+    @Input() code: any = {
+        mode: 'text',
+        options: {}
+    };
+    @Input() height: string;
+    @Input() show = [];
+    @Input() value = '';
+    @Input() width: string;
+
+    @ViewChild('ace') ace: AceEditorComponent;
+
+    selectOptions: SelectItem[] = [];
+    selected: string;
+    styles: any;
+
+    propagateChange = (_: any) => {};
+
+    constructor() {
+    }
+
+    ngOnInit() {
+        if (this.show.length) {
+            this.show.forEach(item => {
+                this.selectOptions.push({
```

Comment:
maybe we should validate that item is a valid options, we could compare with the default selectOptions array.